konig-schemagen/src/main/java/io/konig/schemagen/TableShapeGenerator.java

```diff
@@ -95,22 +72,103 @@ private Shape createTableShape(CreateTable createTable, List<ColumnDefinition> c
 	private void addPropertyContraint(Shape shape,  List<ColumnDefinition> columnList) {
 		
 		List<PropertyConstraint> propertyConstraints = new ArrayList<PropertyConstraint>();
-			for(ColumnDefinition column : columnList) {
-				PropertyConstraint pc = new PropertyConstraint();			
-						pc.setPath(new URIImpl(tableConfig.getPropertyNamespace()+column.getColumnName()));
-						pc.setMaxCount(1);
-						pc.setMinCount(getMinCount(column));
-						pc.setFormula(null);
-						if(("ID").equals(column.getColumnName())){
-							pc.setDatatype(XMLSchema.ANYURI);
-						}else{
-						pc.setDatatype(getDataType(column));
-						}
-						shape.add(pc);
-						propertyConstraints.add(pc);
-				}
+		for (ColumnDefinition column : columnList) {
+			PropertyConstraint pc = new PropertyConstraint();
+			pc.setPath(new URIImpl(tableConfig.getPropertyNamespace() + column.getColumnName()));
+			pc.setMaxCount(1);
+			pc.setMinCount(getMinCount(column));
+			pc.setMaxLength(getMaxLength(column));
+			pc.setMinLength(0);
+			if (("ID").equals(column.getColumnName())) {
+				pc.setDatatype(XMLSchema.ANYURI);
+			} else {
+				pc.setDatatype(getMySqlDataType(column));
 			}
-	private URI getDataType(ColumnDefinition column){
+			if("DOUBLE".equals(column.getColDataType().getDataType()) 
+					&& (getDecimalPrecision(column.getColDataType())!=0)){
+			pc.setDecimalPrecision(getDecimalPrecision(column.getColDataType()));
+			pc.setDecimalScale(getDecimalScale(column.getColDataType()));
+			pc.setDatatype(XMLSchema.DECIMAL);
+			}
+			if ("SIGNED TINYINT".equals(column.getColDataType().getDataType())) {
```

Comment:
Do we know for sure that `column.getColDataType().getDataType()` will always return a single space between words?  Can we prove that there is no other possible whitespace?  

If it is not provable, then you need to normalize the value before making the comparison.

Also, to make the code more readable, it would be better to use a switch statement instead of a long series of `if..then` statements.